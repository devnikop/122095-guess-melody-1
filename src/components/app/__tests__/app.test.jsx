import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import React from "react";
import renderer from "react-test-renderer";

import App from "../app.jsx";

const mock = {
  questions: [
    {
      answers: [
        {
          genre: `rock`,
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        },
        {
          genre: `pop`,
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        },
        {
          genre: `jazz`,
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        },
        {
          genre: `rock`,
          src: `https://upload.wikimedia.org/wikipedia/commons/1/1f/Uganda_flag_and_national_anthem_-_Oh_Uganda_Land_o.ogg`,
        },
      ],
      genre: `rock`,
      type: `genre`,
    },
    {
      answers: [
        {
          picture: `path.jpg`,
          artist: `John Snow`,
        },
        {
          picture: `path.jpg`,
          artist: `Jack Daniels`,
        },
        {
          picture: `path.jpg`,
          artist: `Jim Beam`,
        },
      ],
      song: {
        artist: `Jim Beam`,
        src: `path.mp3`,
      },
      type: `artist`,
    },
  ],
};

const initialState = {
  mistakes: 0,
  step: -1,
};

const genreScreenState = {
  mistakes: 0,
  step: 0
};

const artistScreenState = {
  mistakes: 0,
  step: 1,
};

const middlewares = [];
const mockStore = configureStore(middlewares);

const createNodeMock = (element) => {
  if (element.type === `audio`) {
    return {};
  }
}

describe(`app snapshots:`, () => {
  it(`render welcome-screen with initial state`, () => {
    const { questions } = mock;
    const store = mockStore(initialState);

    const tree = renderer.create(
      <Provider store={store}>
        <App
          errorCount={5}
          gameTime={3}
          questions={questions}
        />
      </Provider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`render genre-screen with updated state`, () => {
    const { questions } = mock;
    const store = mockStore(genreScreenState);

    const tree = renderer.create(
      <Provider store={store} >
        <App
          errorCount={5}
          gameTime={3}
          questions={questions}
        />
      </Provider>,
      { createNodeMock }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`render artist-screen with updated state`, () => {
    const { questions } = mock;
    const store = mockStore(artistScreenState);

    const tree = renderer.create(
      <Provider store={store}>
        <App
          errorCount={5}
          gameTime={3}
          questions={questions}
        />
      </Provider>,
      { createNodeMock }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
