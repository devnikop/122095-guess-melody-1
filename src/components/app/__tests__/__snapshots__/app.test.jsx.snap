// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app snapshots: render artist-screen with updated state 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx={390}
        cy={390}
        r={370}
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="correct"
      />
      <div
        className="correct"
      />
      <div
        className="correct"
      />
    </div>
  </header>
  <section
    className="game game--artist"
  >
    <section
      className="game__screen"
    >
      <h2
        className="game__title"
      >
        Кто исполняет эту песню?
      </h2>
      <div
        className="game__track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
      <form
        className="game__artist"
        onChange={[Function]}
      >
        <div
          className="artist"
        >
          <input
            className="artist__input visually-hidden"
            id="answer-0"
            name="answer"
            type="radio"
            value="answer-0"
          />
          <label
            className="artist__name"
            htmlFor="answer-0"
          >
            <img
              alt="John Snow"
              className="artist__picture"
              src="path.jpg"
            />
            John Snow
          </label>
        </div>
        <div
          className="artist"
        >
          <input
            className="artist__input visually-hidden"
            id="answer-1"
            name="answer"
            type="radio"
            value="answer-1"
          />
          <label
            className="artist__name"
            htmlFor="answer-1"
          >
            <img
              alt="Jack Daniels"
              className="artist__picture"
              src="path.jpg"
            />
            Jack Daniels
          </label>
        </div>
        <div
          className="artist"
        >
          <input
            className="artist__input visually-hidden"
            id="answer-2"
            name="answer"
            type="radio"
            value="answer-2"
          />
          <label
            className="artist__name"
            htmlFor="answer-2"
          >
            <img
              alt="Jim Beam"
              className="artist__picture"
              src="path.jpg"
            />
            Jim Beam
          </label>
        </div>
      </form>
    </section>
  </section>
</section>
`;

exports[`app snapshots: render genre-screen with updated state 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx={390}
        cy={390}
        r={370}
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="correct"
      />
      <div
        className="correct"
      />
      <div
        className="correct"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Выберите 
      rock
       треки
    </h2>
    <form
      className="game__tracks"
      onSubmit={[Function]}
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            defaultValue="answer-0"
            id="answer-0"
            name="answer"
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-0"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            defaultValue="answer-1"
            id="answer-1"
            name="answer"
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-1"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            defaultValue="answer-2"
            id="answer-2"
            name="answer"
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-2"
          >
            Отметить
          </label>
        </div>
      </div>
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
        <div
          className="game__answer"
        >
          <input
            className="game__input visually-hidden"
            defaultValue="answer-3"
            id="answer-3"
            name="answer"
            type="checkbox"
          />
          <label
            className="game__check"
            htmlFor="answer-3"
          >
            Отметить
          </label>
        </div>
      </div>
      <button
        className="game__submit button"
        type="submit"
      >
        Ответить
      </button>
    </form>
  </section>
</section>
`;

exports[`app snapshots: render welcome-screen with initial state 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx={390}
        cy={390}
        r={370}
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="correct"
      />
      <div
        className="correct"
      />
      <div
        className="correct"
      />
    </div>
  </header>
  <section
    className="welcome"
  >
    <div
      className="welcome__logo"
    >
      <img
        alt="Угадай мелодию"
        height="83"
        src="img/melody-logo.png"
        width="186"
      />
    </div>
    <button
      className="welcome__button"
      onClick={[Function]}
    >
      <span
        className="visually-hidden"
      >
        Начать игру
      </span>
    </button>
    <h2
      className="welcome__rules-title"
    >
      Правила игры
    </h2>
    <p
      className="welcome__text"
    >
      Правила просты:
    </p>
    <ul
      className="welcome__rules-list"
    >
      <li>
        За 
        3
         минут нужно ответить на все вопросы.
      </li>
      <li>
        Можно допустить 
        5
         ошибки.
      </li>
    </ul>
    <p
      className="welcome__text"
    >
      Удачи!
    </p>
  </section>
</section>
`;
